contents/docs/cdp/batch-exports/databricks.md

```diff
@@ -0,0 +1,239 @@
+| created_at                 | `TIMESTAMP` | The timestamp when the person was created                                                                      |
+
+The Databricks table will contain one row per `(team_id, distinct_id)` pair, and each pair is mapped to their corresponding `person_id` and latest `properties`.
+
+
+## FAQ
+
+### How does the batch export process work?
+
+The batch export process follows a consistent pattern regardless of the model being exported, with variations depending on whether the model is mutable (like persons) or immutable (like events).
+
+**For immutable models (e.g. events):**
+
+1. Create a temporary volume to store Parquet files.
+2. Create the destination table if it doesn't exist.
+3. Query data from PostHog and transform it into Parquet format.
+4. Upload Parquet files to the temporary volume.
+5. Copy data directly from the volume into the destination table using Databricks' `COPY INTO` command.
+6. Delete the temporary volume.
+
+**For mutable models (e.g. persons, sessions):**
+
+1. Create a temporary volume to store Parquet files.
+2. Create the destination table if it doesn't exist.
+3. Create a temporary stage table with the same schema as the destination table.
+4. Query data from PostHog and transform it into Parquet format.
+5. Upload Parquet files to the temporary volume.
+6. Copy data from the volume into the **stage table** using `COPY INTO`.
+7. Execute a `MERGE` operation between the stage table and the destination table:
+   - Update existing rows where the stage table has newer versions.
+   - Insert new rows that don't exist in the destination table.
+8. Delete the stage table and temporary volume.
+
+The merge operation for mutable models ensures that:
+- Rows that match in the destination table and have newer version fields in the stage table are updated.
+- New rows not found in the destination table are inserted.
+- Your data always reflects the latest state while maintaining historical records that haven't been updated.
+
+### What is the benefit of table partitioning?
+
+Partitioning your Databricks table can significantly improve query performance, especially for large datasets. When you partition by a field like `timestamp`, Databricks can skip reading entire partitions of data when you filter by that field in your queries.
+
+For example, if you partition by `timestamp` and run a query filtering for events from the last 7 days, Databricks will only scan the relevant partitions rather than the entire table.
+
+By default, PostHog automatically partitions:
+- Events tables by `timestamp`
+- Sessions tables by `end_timestamp`
+
+You can override this by specifying a different partition field in the configuration.
+
+### Should I use VARIANT or STRING type for JSON fields?
+
+We recommend using `VARIANT` (the default) for the following reasons:
+- Better query performance for nested JSON data
+- More efficient storage
+- Type safety and validation
```

Comment:
How does using variant type provide type safety and validation? the json can still be garbage.

Reply:
yep, have removed since it was confusing